src/main/java/com/android/volley/toolbox/CacheHeader.java

```diff
@@ -115,6 +115,36 @@ static CacheHeader readHeader(DiskBasedCache.CountingInputStream is) throws IOEx
                 key, etag, serverDate, lastModified, ttl, softTtl, allResponseHeaders);
     }
 
+    /**
+     * Reads the header from a AsynchronousFileChannel and returns a CacheHeader object.
+     *
+     * @param buffer Buffer to get header info from.
+     * @throws IOException if fails to read header
+     */
+    @Nullable
+    static CacheHeader readHeader(final ByteBuffer buffer) {
+        try {
+            buffer.flip();
+            int magic = buffer.getInt();
```

Comment:
Just to check, should I catch all runtime exceptions or just a BufferUnderflowException?

Reply:
Just BufferUnderflow - anything else would be an unexpected/developer error, even in the face of invalid data. (We should also make sure we handle invalid UTF-8 data when reading strings in case that results in a runtime exception rather than just garbage data, though).